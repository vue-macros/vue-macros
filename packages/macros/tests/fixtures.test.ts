import { resolve } from 'node:path'
import {
  rollupBuild,
  RollupVue,
  RollupVueJsx,
  testFixtures,
  UnpluginOxc,
} from '@vue-macros/test-utils'
import { describe } from 'vitest'
import VueMacros from '../src/rollup'

describe('fixtures', async () => {
  await testFixtures(
    ['tests/fixtures/**/*.{vue,js,jsx,ts,tsx}', '!tests/fixtures/types.ts'],
    (args, id) => {
      return rollupBuild(id, [
        VueMacros({
          setupSFC: true,
          version: 3,
          plugins: {
            vue: RollupVue({
              include: [/\.vue$/, /\.setup\.[cm]?[jt]sx?/],
            }),
            vueJsx: RollupVueJsx(),
          },
        }),
        UnpluginOxc.rollup(),
      ])
    },
    {
      cwd: resolve(__dirname, '..'),
      promise: true,
    },
  )
})
