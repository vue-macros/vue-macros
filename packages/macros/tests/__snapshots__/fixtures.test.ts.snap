// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fixtures > tests/fixtures/basic.vue 1`] = `
"// basic.js
import { defineComponent } from 'vue';
import __MACROS_useVModel from '/vue-macros/define-models/use-vmodel';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

const name = "Foo";
const __default__ = defineComponent({ name });
var _sfc_main = /* @__PURE__ */ defineComponent({
	...__default__,
	props: {
		title: {
			type: String,
			required: true
		},
		modelValue: {
			type: String,
			required: true
		}
	},
	emits: ["change", "update:modelValue"],
	setup(__props, { expose: __expose, emit: __emit }) {
		__expose();
		const emit = __emit;
		let { modelValue } = __MACROS_useVModel("modelValue");
		const handleClick = () => {
			emit("change");
			modelValue.value = "hello, " + __props.title;
		};
		const __returned__ = {
			name,
			emit,
			get modelValue() {
				return modelValue;
			},
			set modelValue(v) {
				modelValue = v;
			},
			handleClick,
			__MACROS_defineComponent: defineComponent,
			get __MACROS_useVModel() {
				return __MACROS_useVModel;
			}
		};
		Object.defineProperty(__returned__, "__isScriptSetup", {
			enumerable: false,
			value: true
		});
		return __returned__;
	}
});

var basic = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { basic as default };
"
`;

exports[`fixtures > tests/fixtures/boolean-prop.vue 1`] = `
"// boolean-prop.js
import { resolveComponent, createElementBlock, openBlock, Fragment, createCommentVNode, createVNode } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

const _sfc_main = {  };

function _sfc_render(_ctx, _cache) {
  const _component_el_form = resolveComponent("el-form");

  return (openBlock(), createElementBlock(Fragment, null, [
    createCommentVNode(" disabled "),
    createVNode(_component_el_form, { inline: "" })
  ], 2112 /* STABLE_FRAGMENT, DEV_ROOT_FRAGMENT */))
}
var booleanProp = /*#__PURE__*/_export_sfc(_sfc_main, [['render',_sfc_render],__FILE__]);

export { booleanProp as default };
"
`;

exports[`fixtures > tests/fixtures/case-233.vue 1`] = `
"// case-233.js
import { defineComponent } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

const msg = "Hello";
var _sfc_main = /* @__PURE__ */ defineComponent({
	__name: "case-233",
	props: { msg: {
		type: String,
		required: true,
		default: msg
	} },
	setup(__props, { expose: __expose }) {
		__expose();
		const __returned__ = { msg };
		Object.defineProperty(__returned__, "__isScriptSetup", {
			enumerable: false,
			value: true
		});
		return __returned__;
	}
});

var case233 = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { case233 as default };
"
`;

exports[`fixtures > tests/fixtures/define-props-refs.vue 1`] = `
"// define-props-refs.js
import { defineComponent, toRefs } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
	__name: "define-props-refs",
	props: { foo: {
		type: String,
		required: true
	} },
	setup(__props, { expose: __expose }) {
		__expose();
		const __MACROS_props = __props;
		const { foo } = toRefs(__MACROS_props);
		console.log(foo.value);
		// @ts-expect-error
		foo.value = "hello";
		const __returned__ = {
			__MACROS_props,
			foo,
			__MACROS_toRefs: toRefs
		};
		Object.defineProperty(__returned__, "__isScriptSetup", {
			enumerable: false,
			value: true
		});
		return __returned__;
	}
});

var definePropsRefs = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { definePropsRefs as default };
"
`;

exports[`fixtures > tests/fixtures/error-diff-lang.vue 1`] = `[Error: [vue-macros] <script> and <script setup> must have the same language type.]`;

exports[`fixtures > tests/fixtures/hoist-static.vue 1`] = `
"// hoist-static.js
import { defineComponent } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

const name = 'App';
const propName = 'title';
const __default__ = defineComponent({
  name,
});
const _sfc_main = /*@__PURE__*/Object.assign(__default__, {
  props: [propName],
  setup(__props, { expose: __expose }) {
  __expose();







const __returned__ = { name, propName, __MACROS_defineComponent: defineComponent };
Object.defineProperty(__returned__, '__isScriptSetup', { enumerable: false, value: true });
return __returned__
}

});
var hoistStatic = /*#__PURE__*/_export_sfc(_sfc_main, [__FILE__]);

export { hoistStatic as default };
"
`;

exports[`fixtures > tests/fixtures/hoist-static-scope.vue 1`] = `
"// hoist-static-scope.js
import { defineComponent, createElementBlock, openBlock } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

const [name] = ["VarCompName"];
const __default__ = defineComponent({ name });
var _sfc_main = /* @__PURE__ */ defineComponent({
	...__default__,
	setup(__props, { expose: __expose }) {
		__expose();
		// https://github.com/vue-macros/vue-macros/issues/37#issuecomment-1231677842
		const test = (name) => name;
		const __returned__ = {
			name,
			test
		};
		Object.defineProperty(__returned__, "__isScriptSetup", {
			enumerable: false,
			value: true
		});
		return __returned__;
	}
});

function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock("div", null, "1");
}
var hoistStaticScope = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render], __FILE__]);

export { hoistStaticScope as default };
"
`;

exports[`fixtures > tests/fixtures/mixed-define-props.vue 1`] = `
"// mixed-define-props.js
import { defineComponent, toRefs, toRef } from 'vue';
import __MACROS_normalizePropsOrEmits from '/vue-macros/define-prop/helper';
import __MACROS_useVModel from '/vue-macros/define-models/use-vmodel';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
	__name: "mixed-define-props",
	props: {
		...__MACROS_normalizePropsOrEmits({
			foo: {
				type: String,
				required: false
			},
			baz: {
				type: String,
				required: true
			}
		}),
		...__MACROS_normalizePropsOrEmits(["bar"])
	},
	emits: ["update:baz"],
	setup(__props, { expose: __expose }) {
		__expose();
		const __MACROS_props = __props;
		const { foo } = toRefs(__MACROS_props);
		const bar = toRef(__props, "bar");
		const { baz } = __MACROS_useVModel("baz");
		const __returned__ = {
			__MACROS_props,
			foo,
			bar,
			baz,
			__MACROS_toRef: toRef,
			get __MACROS_normalizePropsOrEmits() {
				return __MACROS_normalizePropsOrEmits;
			},
			get __MACROS_useVModel() {
				return __MACROS_useVModel;
			},
			__MACROS_toRefs: toRefs
		};
		Object.defineProperty(__returned__, "__isScriptSetup", {
			enumerable: false,
			value: true
		});
		return __returned__;
	}
});

var mixedDefineProps = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { mixedDefineProps as default };
"
`;

exports[`fixtures > tests/fixtures/short-emits.setup.tsx 1`] = `
"// short-emits.setup.js
import { defineComponent } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
	__name: "short-emits.setup",
	emits: ["update:modelValue"],
	setup(__props, { expose: __expose }) {
		__expose();
		const __returned__ = {};
		Object.defineProperty(__returned__, "__isScriptSetup", {
			enumerable: false,
			value: true
		});
		return __returned__;
	}
});

var shortEmits_setup = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { shortEmits_setup as default };
"
`;
